### ISSUE TITLE
Can I set the gpt2 inference model text generation length?

### ISSUE THREAD

**Issue by YINGPENGZH** [bytedance/lightseq#223]
The following code is part of the example, I want to set the generation length when using gpt2. For instance, my prefix is "My name is GPT", which I want to generate six words, it may generate result: "My name is GPT which is a text generation model. " And not generate seventh, eighth words and more. Cound it lightseq support?

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("--user_input", action="store_true")
    args = parser.parse_args()
    ls_tokenizer = GPT2Tokenizer.from_pretrained("gpt2")
    ls_tokenizer.add_special_tokens({"pad_token": "[PAD]"})
    ls_model = lsi.Gpt("lightseq_gpt2_base.hdf5", max_batch_size=16)  // this code which i want to control  the length of generation

**Comment by neopro12** [comment]
Essentially, beam search or greedy search only takes the current step into account. 
Generation with fixed length needs a new search algorithm rather than implementation.

**Comment by YINGPENGZH** [comment]
> Essentially, beam search or greedy search only takes the current step into account. Generation with fixed length needs a new search algorithm rather than implementation.

Yes, I want to control the beam search counts: I use pytorch & transformers which can control the steps of beam search. Take the following pseudo code for instance，can lightseq inference like this? Or how lightseq judge the end of every sequance: 
**prefix = "My name is GPT"   # raw data
prefix_id =  tokenizer.convert_tokens_to_ids(prefix)  #convert to id
prefix = torch.tensor(context, dtype=torch.long, device=device)  
generated = prefix
for _ in trange(length):     # the length i want to generate
       inputs = {'input_ids': generated}
       outputs = model(**inputs)  
       next_token = beam_search(outputs)    # the next token gpt2 generate
       generated = torch.cat((generated, next_token), dim=-1)**   # concat the next token
